pkgs/applications/video/davinci-resolve/default.nix

```diff
@@ -252,7 +256,28 @@ buildFHSEnv {
     ''
   }";
 
-  passthru = { inherit davinci; };
+  passthru = {
+    inherit davinci;
+    updateScript = lib.getExe (writeShellApplication {
+      name = "update-davinci-resolve";
+      runtimeInputs = [ curl jq yq common-updater-scripts ];
+      text = ''
+        set -o errexit
+        drv=pkgs/applications/video/davinci-resolve/default.nix
+        currentVersion="$(grep -E 'version = "[0-9]+\.[0-9]+\.[0-9]+"' $drv | cut -d '"' -f2)"
+        downloadsJSON="$(curl -s https://www.blackmagicdesign.com/api/support/us/downloads.json)"
+
+        latestLinuxVersion="$(jq '[.downloads[] | select(.urls.Linux) | .urls.Linux[] | select(.downloadTitle | test("DaVinci Resolve")) | .downloadTitle]' <(echo "$downloadsJSON") | grep -oP 'DaVinci Resolve \K\d+\.\d+\.\d+' | sort | tail -n 1)"
+        update-source-version davinci-resolve "$latestLinuxVersion" --source-key=davinci.src
+
+        # Since the standard and studio both use the same version we need to reset it before updating studio
+        sed -i -e "s/""$latestLinuxVersion""/""$currentVersion""/" "$drv"
+
+        latestStudioLinuxVersion="$(jq '[.downloads[] | select(.urls.Linux) | .urls.Linux[] | select(.downloadTitle | test("DaVinci Resolve")) | .downloadTitle]' <(echo "$downloadsJSON") | grep -oP 'DaVinci Resolve Studio \K\d+\.\d+\.\d+' | sort | tail -n 1)"
+        update-source-version davinci-resolve-studio "$latestStudioLinuxVersion" --source-key=davinci.src
```

Comment:
@amarshall  This is the only change I'm not sure about.

Since `studio` is so tightly coupled with the standard version I'm hesitant to assume Blackmagic will always release both versions at the same time. If there is a situation where only one of the two packages is bumped a version we might be stuck with the older of the two. I prefer we verify each individually.

But, I'm not entirely sure how the CI would handle one of the two failing.. Might produce an interesting PR we would have to dismiss but figure we can cross that bridge when we get there haha. 

Not set on this though, just my thoughts.

Reply:
I think we can just roll with it and see what happens heh.